bftengine/src/communication/TlsTCPCommunication.cpp

```diff
@@ -326,32 +288,35 @@ class AsyncTlsConnection : public enable_shared_from_this<AsyncTlsConnection> {
                          string connectionType,
                          string subject,
                          NodeNum expectedPeerId = -1) {
+
+    // first, basic sanity test, just to eliminate disk read if the certificate
+    // is unknown
     regex r("OU=\\d*", regex_constants::icase);
     smatch sm;
     regex_search(subject, sm, r);
     if (4 > sm.length()) {
-      LOG_ERROR("OU not found or empty: " << subject);
+      LOG_ERROR(_logger, "OU not found or empty: " << subject);
       return false;
     }
 
-    string remPeer = sm.str().substr(0, sm.str().length() - 3);
+    string remPeer = sm.str().substr(3, sm.str().length() - 3);
```

Comment:
Given that it looks like there was a bug here, a comment about what this "3" is would be a nice addition.

Reply:
Fixed